oak_attestation_verification/src/amd.rs

```diff
@@ -0,0 +1,199 @@
+}
+
+fn chip_id(cert: &Certificate) -> anyhow::Result<[u8; 64]> {
+    let exts = cert
+        .tbs_certificate
+        .extensions
+        .as_ref()
+        .ok_or_else(|| anyhow::anyhow!("could not get extensions from cert"))?;
+    let chip_id_ext = exts
+        .iter()
+        .find(|&ext| ext.extn_id == CHIP_ID_OID)
+        .ok_or_else(|| anyhow::anyhow!("no chip ID found in cert"))?;
+    let chip_id = chip_id_ext.extn_value.as_bytes().to_vec();
+    anyhow::ensure!(chip_id.len() == 64, "length of chip ID value is not 64");
+
+    // Copy into array of fixed length.
+    let mut result = [0; 64];
+    result.copy_from_slice(&chip_id[0..64]);
+    Ok(result)
+}
+
+fn tcb_version(vcek: &Certificate) -> anyhow::Result<TcbVersion> {
+    let mut tcb = TcbVersion::new_zeroed();
+    for ext in vcek
+        .tbs_certificate
+        .extensions
+        .as_ref()
+        .ok_or_else(|| anyhow::anyhow!("could not get extensions from VCEK cert"))?
+    {
+        // What's the appropriate way of extracting u8 from OctetString?
+        let arr = ext.extn_value.as_bytes();
+        let last = arr.len() - 1;
+        if ext.extn_id == BL_OID {
+            tcb.boot_loader = arr[last];
+        } else if ext.extn_id == TEE_OID {
+            tcb.tee = arr[last];
+        } else if ext.extn_id == SNP_OID {
+            tcb.snp = arr[last];
+        } else if ext.extn_id == UCODE_OID {
+            tcb.microcode = arr[last];
+        }
+    }
+    Ok(tcb)
+}
+
+pub fn verify_attestation_report_signature(
+    vcek: &Certificate,
+    report: &AttestationReport,
+) -> anyhow::Result<()> {
+    // First check some necessary conditions for the signature to be valid.
+    let arpt_chip_id = report.data.chip_id;
+    let vcek_chip_id = chip_id(vcek)?;
+    anyhow::ensure!(
+        arpt_chip_id == vcek_chip_id,
+        "chip id differs attestation={} vcek={}",
+        hex::encode(arpt_chip_id),
+        hex::encode(vcek_chip_id)
+    );
+
+    let arpt_tcb = &report.data.reported_tcb;
+    let vcek_tcb = tcb_version(vcek)?;
+    anyhow::ensure!(
+        arpt_tcb.snp == vcek_tcb.snp
+            && arpt_tcb.microcode == vcek_tcb.microcode
+            && arpt_tcb.tee == vcek_tcb.tee
+            && arpt_tcb.boot_loader == vcek_tcb.boot_loader,
+        "mismatch in TCB version"
+    );
+
+    let verifying_key = {
+        let pubkey_info = vcek.tbs_certificate.subject_public_key_info.owned_to_ref();
+        p384::ecdsa::VerifyingKey::from_sec1_bytes(pubkey_info.subject_public_key.raw_bytes())
+            .map_err(|_err| anyhow::anyhow!("could not extract ECDSA P-384 public key"))
+    }?;
+    let message = report.data.as_bytes();
+
+    // `report.signature.{r,s}` contain 48 bytes, the rest is zero padded
+    // to allow longer signatures as well. The 48 bytes are interpreted as
+    // a single little-endian encoded integer. `p384::ecdsa::Signature`
+    // requires big-endian, so need to mirror.
```

Comment:
Could you use an endianness conversion function instead of doing the mirroring manually here?